packages/flutter/lib/src/services/text_input.dart

```diff
@@ -461,24 +461,16 @@ class TextInputConfiguration {
     SmartDashesType? smartDashesType,
     SmartQuotesType? smartQuotesType,
     this.enableSuggestions = true,
+    this.enableInteractiveSelection = true,
     this.actionLabel,
     this.inputAction = TextInputAction.done,
     this.keyboardAppearance = Brightness.light,
     this.textCapitalization = TextCapitalization.none,
     this.autofillConfiguration = AutofillConfiguration.disabled,
     this.enableIMEPersonalizedLearning = true,
     this.enableDeltaModel = false,
-  }) : assert(inputType != null),
```

Comment:
these are all non-null type, these checks are redundant

Reply:
Good point, I thought that the compiler wouldn't let you do this, but I guess it will.  I'll keep an eye out for other places that can be cleaned up like this.

Reply:
It looks like it will still be useful for user that has not migrate to null safety, so I  will revert this change

Reply:
Ah that's right.  Sounds good.